piff/gp_interp_2pcf.py

```diff
@@ -0,0 +1,375 @@
+# Copyright (c) 2016 by Mike Jarvis and the other collaborators on GitHub at
+# https://github.com/rmjarvis/Piff  All rights reserved.
+#
+# Piff is free software: Redistribution and use in source and binary forms
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+.. module:: gp_interp_2pcf
+"""
+
+import numpy as np
+import treecorr
+import copy
+
+from sklearn.gaussian_process.kernels import Kernel
+import scipy.optimize as op
+from scipy.linalg import cholesky, cho_solve
+
+from .interp import Interp
+from .star import Star, StarFit
+
+
+class GPInterp2pcf(Interp):
```

Comment:
So based on the fact that this is a sort of specialization of GPInterp, I am wondering if it shouldn't inherit from that instead. Maybe you have to change enough functions that this is clearer, but as I go through here I do find myself wondering whether some of the changes here are backported back to GPInterp.

Reply:
I guess, I changed to much things to make the possibility that GPInterp2pcf inherit from GPInterp. If I would changed only the strategy of searching hyperparameter, It would be ok. However, the fact that I am fitting different hyperparameter(s) per PSF’s parameter(s) changes a lot of thing within the class. That should be backported to GPinterp.